src/ring_theory/power_series/basic.lean

```diff
@@ -1178,6 +1178,27 @@ lemma mul_inv_of_unit (φ : power_series R) (u : units R) (h : constant_coeff R
   φ * inv_of_unit φ u = 1 :=
 mv_power_series.mul_inv_of_unit φ u $ h
 
+/-- Two ways of removing the constant coefficient of a power series are the same. -/
+lemma sub_const_eq_shift_mul_X (φ : power_series R) :
+  φ - C R (constant_coeff R φ) = power_series.mk (λ p, coeff R (p + 1) φ) * X :=
```

Comment:
Is it worth proving this for `semiring` too, as:
```lean
lemma eq_shift_mul_X_add_const (φ : power_series R) :
  φ = power_series.mk (λ p, coeff R (p + 1) φ) * X + C R (constant_coeff R φ) :=
```
Your current proofs would follow trivially from this

Reply:
Good point! I had thought about it, but then I sort the semiring case was too exotic. I'm glad it's not..
Added the lemmas there and use them in the proof below.